### PR TITLE
Add katib application CR

### LINKED ISSUE #117: Katib needs an Application CR
The katib kustomize package doesn't currently contain an Application custom resource.
https://github.com/kubeflow/manifests/tree/master/katib/base

We should add an Application CR to provide meta information.

### PR DESCRIPTION
**Pull request by wuchunghsuan**
**Which issue is resolved by this Pull Request:**
Resolves #117 

**Description of your changes:**
Add katib application CR

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/225)

### CODE REVIEW

File: katib-v1alpha2/katib-manager/overlays/application/application.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      app.kubernetes.io/version: v0.6
+  componentKinds:
+  - group: core
+    kind: Service
+  - group: apps
+    kind: Deployment
+  descriptor:
+    type: "katib-manager"
+    version: "v1alpha2"
+    description: "Katib manager allows users to launch AutoML jobs on katib and manage them."
```
Comment (richardsliu) [inline katib-v1alpha2/katib-manager/overlays/application/application.yaml]: Katib manager doesn't launch jobs; it is more like a storage API layer.

File: katib-v1alpha2/katib-controller/overlays/application/application.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - group: kubeflow.org
+    kind: Experiment
+  - group: kubeflow.org
+    kind: Trial
+  descriptor:
+    type: "katib-controller"
+    version: "v1alpha2"
+    description: "Katib controller is used to create custom resource \"Experiment\" and \"Trial\"."
+    maintainers:
+    - name: Zhongxuan Wu
```
[the remark this answers is not in the document]
Comment (wuchunghsuan) [reply katib-v1alpha2/katib-controller/overlays/application/application.yaml]: done

File: katib-v1alpha2/katib-db/overlays/application/application.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  - group: core
+    kind: Secret
+  - group: core
+    kind: PersistentVolumeClaim
+  descriptor:
+    type: "katib-db"
+    version: "v1alpha2"
+    description: "Katib db provides the storage service to the katib manager."
+    maintainers:
+    - name: Zhongxuan Wu
```
[the remark this answers is not in the document]
Comment (wuchunghsuan) [reply katib-v1alpha2/katib-db/overlays/application/application.yaml]: done

File: katib-v1alpha2/katib-manager/overlays/application/application.yaml (hunk 1 above)
Comment (wuchunghsuan) [reply katib-v1alpha2/katib-manager/overlays/application/application.yaml]: I changed the description.

File: katib-v1alpha2/katib-controller/overlays/application/application.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      email: wuzhongxuan@caicloud.io
+    owners:
+    - name: Zhongxuan Wu
+      email: wuzhongxuan@caicloud.io
+    keywords:
+    - katib
+    - katib-controller
+    - AutoML
```
[the remark this answers is not in the document]
Comment (wuchunghsuan) [reply katib-v1alpha2/katib-controller/overlays/application/application.yaml]: done

**Comment by gaocegege** [comment]
@wuchunghsuan 

Thanks for your contribution! :tada: :+1:

**Pull request merged**